tests/service/predictors/coref_test.py

```diff
@@ -0,0 +1,32 @@
+# pylint: disable=no-self-use,invalid-name
+from unittest import TestCase
+
+from allennlp.models.archival import load_archive
+from allennlp.service.predictors import Predictor
+
+
+class TestCorefPredictor(TestCase):
+
+    def test_uses_named_inputs(self):
+
+        inputs = {"document": "This is a single string document about a test. Sometimes it "
+                              "contains coreferent parts."}
+        archive = load_archive('tests/fixtures/coref/serialization/model.tar.gz')
+        predictor = Predictor.from_archive(archive, 'coreference-resolution')
+        result = predictor.predict_json(inputs)
+
+        document = result["document"]
+        assert document == ['This', 'is', 'a', 'single', 'string',
+                            'document', 'about', 'a', 'test', '.', 'Sometimes',
+                            'it', 'contains', 'coreferent', 'parts', '.']
+
+        clusters = result["clusters"]
+        assert isinstance(clusters, list)
+
+        for cluster in clusters:
+            # Spans should be integer indices.
+            assert isinstance(cluster[0], int)
```

Comment:
I'm confused that this passes.  Shouldn't you get a list of clusters, where each cluster is a list of mentions?  There should be another nesting here, I think:

```python
for cluster in clusters:
    for mention in cluster:
        assert isinstance(mention[0], int)
        ...
```

Reply:
Yeah thanks, I always forget about the `output = output[0]` in `forward_on_instance`.